### PR TITLE
Added a new interface to allow to provide FOSHttpCache event subscribers

### PR DESCRIPTION
**Pull request by Toflar**

### CODE REVIEW

File: src/HttpCache/FOSHttpCacheSubscriberPluginInterface.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * You can use this interface to provide your own FOSHttpCache event subscribers.
+ */
+interface FOSHttpCacheSubscriberPluginInterface
+{
+    /**
+     * Gets an array of subscribers.
+     *
+     * @return EventSubscriberInterface[]
+     */
+    public function getSubscribers();
```
Comment (Toflar) [inline src/HttpCache/FOSHttpCacheSubscriberPluginInterface.php:29]: Maybe we need a more distinctive method name here? @aschempp

Comment (leofeyer) [reply src/HttpCache/FOSHttpCacheSubscriberPluginInterface.php:29]: Probably `getCacheSubscribers()`?

**Comment by Toflar** [comment]
@aschempp and I were also discussing about having a general plugin that has access to the `Kernel` and the `HttpCache` (which can be `null` of course). That would be a lot more powerful but at the same time we don't know if we should allow that. Does anybody see any downsides?
@contao/developers ? Also pinging @dmolineus because I know he's doing loads of Contao stuff and has also worked with FOSHttpCache.

**Comment by Toflar** [comment]
This PR must not be merged until @aschempp finished working on his alternative entrypoint handling in https://github.com/contao/contao/pull/152.

**Pull request closed without merging**